### ISSUE TITLE
Version conflict with imported packages

### ISSUE THREAD

**Issue by ryanbrandenburg** [microsoft/msbuild#1512]
In https://github.com/aspnet/MvcPrecompilation/pull/49 we discovered that if you try to run a task which uses a conflicting version of packages like CoreFx then the task will crash with errors like `System.IO.FileNotFoundException: Could not load file or assembly 'System.Collections.Immutable, Version=1.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.`

@pranavkm has created a repro app [here](https://github.com/pranavkm/msbuild-repro/tree/master/src).

**Comment by paul1956** [comment]
This also happens with Visual Studio VSIX projects that use
```
    Public cache As New WeakReference(Of ImmutableDictionary(Of SyntaxTree, Boolean))(ImmutableDictionary(Of SyntaxTree, Boolean).Empty)
```

**Comment by paul1956** [comment]
@ryanbrandenburg did you find any workaround? It is always that exact error.

**Comment by pranavkm** [comment]
@paul1956 we did not dig in to this issue further. For precompilation, we went with an alternate route and spawned a new dotnet process (via `Exec`).

**Comment by rainersigwald** [comment]
Fixed by #4916
